### PR TITLE
[BEAM-5681] Fix website tasks when pull-request ID is specified

### PR DESCRIPTION
**Pull request by swegner**
Website gradle tasks have special support for when running within the context of a PR, in order to stage files on GCS. However, adding this support broke the non-PR mode along the way.

Specifically, using a PR-based `baseUrl` would break the http-proofer internal URL checks, because it would expect to find resources at a /1234/ prefix path. The fix for this is to actually generate content in that prefixed location.

This PR makes all tasks consistent with how they expect and use baseurl paths.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by swegner** [comment]
R: @alanmyrvold 
CC: @udim 

I'm in the process of validating to ensure that all tasks pass in both PR and non-PR scenarios.

**Comment by swegner** [comment]
Tested:

| X | No PR | `-PgithubPullRequestId=6595` |
|--|--|--|
| `:buildWebsite` | [success](https://gradle.com/s/flruutfkrwj6q) | [success](https://gradle.com/s/urafoimblbgj2) |
| `:serveWebsite` | success | success |
| `:testWebsite` | [success](https://gradle.com/s/ra37ssnz3g7zc) | [**failure**](https://gradle.com/s/zw2w537ax373m) |
| `:commitWebsite` | [success](https://gradle.com/s/644a2nhzszn36) | |

**Comment by swegner** [comment]
Latest changes pass manual testing:

| X | No PR | `-PgithubPullRequestId=1234` |
|--|--|--|
| `:testWebsite` | [success](https://gradle.com/s/duwp3if7d327i) | [success](https://gradle.com/s/nnkubb5hft3lk) |
| `:commitWebsite` | [success](https://gradle.com/s/2c57x4t536flk) | [success](https://gradle.com/s/q4iuvhmc44emw) |
| `:stageWebsite` | [success](https://gradle.com/s/itjemkissac74) | [success](https://gradle.com/s/tladagge3rzow) |

File: website/_config.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ description: > # this means to ignore newlines until "baseurl:"
   users to easily implement their data integration processes.
 
 # the subpath of your site, e.g. /blog
-baseurl: ""
+# baseurl: ""
```
Comment (alanmyrvold) [inline website/_config.yml]: Should this be removed instead of commented out?

File: website/build.gradle [hunk 2]
```diff
@@ -230,11 +264,11 @@ publishWebsite.dependsOn commitWebsite
  *   ./gradlew :beam-website:stageWebsite -PgithubPullRequestId=${ghprbPullId} -PwebsiteBucket=foo
  */
 task stageWebsite << {
-  assert project.hasProperty('githubPullRequestId')
-  assert githubPullRequestId.isInteger()
+  // TODO: Refactor this duplication with build task
+  def baseUrl = project.findProperty('githubPullRequestId') ? "${project.findProperty('githubPullRequestId')}" : 'latest'
```
Comment (alanmyrvold) [inline website/build.gradle]: Not sure this works ... I'm seeing the findProperty succeeding in non-PR jobs with the value remaining the uninterpolated value, --baseurl=/${ghprbPullId}

File: website/_config.yml (hunk 1 above)
Comment (swegner) [reply website/_config.yml]: Done.

File: website/build.gradle (hunk 2 above)
Comment (swegner) [reply website/build.gradle]: This works in the latest version. I ran `./gradlew :beam-website:stageWebsite` locally: [Gradle Build Scan](https://gradle.com/s/rmd2mxqfbrofg)

```
Website published to http://apache-beam-website-pull-requests.storage.googleapis.com/latest/index.html
```

**Comment by swegner** [comment]
I believe this is ready to go. @alanmyrvold could you please take a look?

**Pull request merged**